pipeline/wpull_hooks.py

```diff
@@ -27,20 +27,21 @@
 logger = logging.getLogger('archivebot.pipeline.wpull_hooks')
 
 
-def log_ignore(url, pattern):
+def log_ignore(url, pattern, source):
   packet = dict(
     ts=time.time(),
     url=url,
     pattern=pattern,
-    type='ignore'
+    type='ignore',
+    source=source,
   )
 
   control.log(packet, ident, log_key)
 
-
-def maybe_log_ignore(url, pattern):
+# source is a single character identifying what caused the ignore
+def maybe_log_ignore(url, pattern, source='U'):
```

Comment:
I'd prefer more than a single character with no default: 

1. All call sites of `maybe_log_ignore` can be found, so no default is necessary.
2. The extra size of longer `source`s doesn't matter in the default `igoff` mode, and `igon` is not left on for very long anyway.  (Also, ArchiveBot pipelines tend to be run on machines that have access to high-bandwidth pipes.)